code/lib/codemod/src/transforms/csf-2-to-3.ts

```diff
@@ -179,13 +185,122 @@ export default function transform({ source, path }: FileInfo, api: API, options:
     output = prettier.format(output, {
       ...prettierConfig,
       // This will infer the parser from the filename.
-      filepath: path,
+      filepath: info.path,
     });
   } catch (e) {
-    logger.log(`Failed applying prettier to ${path}.`);
+    logger.log(`Failed applying prettier to ${info.path}.`);
   }
 
   return output;
 }
 
+class StorybookImportHelper {
+  constructor(csf: CsfFile, info: FileInfo) {
+    // This allows for showing buildCodeFrameError messages
+    // @ts-expect-error File is not yet exposed, see https://github.com/babel/babel/issues/11350#issuecomment-644118606
+    const file: BabelFile = new babel.File(
+      { filename: info.path },
+      { code: info.source, ast: csf._ast }
+    );
+    this.sbImportDeclarations = this.getAllSbImportDeclarations(file);
+  }
+
+  private sbImportDeclarations: NodePath<t.ImportDeclaration>[];
+
+  private getAllSbImportDeclarations = (file: BabelFile) => {
+    const found: NodePath<t.ImportDeclaration>[] = [];
+
+    file.path.traverse({
+      ImportDeclaration: (path) => {
+        if (!path.node.source.value.startsWith('@storybook')) return;
+        const isRendererImport = path.get('specifiers').some((specifier) => {
+          if (specifier.isImportNamespaceSpecifier()) {
+            throw path.buildCodeFrameError(
+              `This codemod does not support namespace imports for a ${path.node.source.value} package.
+            Replace the namespace import with named imports and try again.`
+            );
+          }
+          if (!specifier.isImportSpecifier()) return false;
+          const imported = specifier.get('imported');
+          if (!imported.isIdentifier()) return false;
+
+          return [
+            'Story',
+            'StoryFn',
+            'StoryObj',
+            'Meta',
+            'ComponentStory',
+            'ComponentStoryFn',
+            'ComponentStoryObj',
```

Comment:
In a followup PR, consider updating this to rename `ComponentStoryObj` to `StoryObj`, which won't happen currently.

Also, ideally consolidate `ComponentX` related logic into a single test case so it's easy to visualize all the possibilities.

Reply:
Will do in the next PR